src/Middleware/OutputCaching/src/LoggerExtensions.cs

```diff
@@ -0,0 +1,128 @@
+        EventName = "NotModifiedIfModifiedSinceSatisfied")]
+    internal static partial void NotModifiedIfModifiedSinceSatisfied(this ILogger logger, DateTimeOffset lastModified, DateTimeOffset ifModifiedSince);
+
+    [LoggerMessage(21, LogLevel.Information, "The content requested has not been modified.", EventName = "NotModifiedServed")]
+    internal static partial void NotModifiedServed(this ILogger logger);
+
+    [LoggerMessage(22, LogLevel.Information, "Serving response from cache.", EventName = "CachedResponseServed")]
+    internal static partial void CachedResponseServed(this ILogger logger);
+
+    [LoggerMessage(23, LogLevel.Information, "No cached response available for this request and the 'only-if-cached' cache directive was specified.",
+        EventName = "GatewayTimeoutServed")]
+    internal static partial void GatewayTimeoutServed(this ILogger logger);
+
+    [LoggerMessage(24, LogLevel.Information, "No cached response available for this request.", EventName = "NoResponseServed")]
+    internal static partial void NoResponseServed(this ILogger logger);
+
+    [LoggerMessage(25, LogLevel.Debug, "Vary by rules were updated. Headers: {Headers}, Query keys: {QueryKeys}", EventName = "VaryByRulesUpdated")]
+    internal static partial void VaryByRulesUpdated(this ILogger logger, string headers, string queryKeys);
+
+    [LoggerMessage(26, LogLevel.Information, "The response has been cached.", EventName = "ResponseCached")]
+    internal static partial void ResponseCached(this ILogger logger);
+
+    [LoggerMessage(27, LogLevel.Information, "The response could not be cached for this request.", EventName = "ResponseNotCached")]
+    internal static partial void LogResponseNotCached(this ILogger logger);
+
+    [LoggerMessage(28, LogLevel.Warning, "The response could not be cached for this request because the 'Content-Length' did not match the body length.",
+        EventName = "responseContentLengthMismatchNotCached")]
+    internal static partial void ResponseContentLengthMismatchNotCached(this ILogger logger);
+
+    [LoggerMessage(29, LogLevel.Debug,
+        "The age of the entry is {Age} and has exceeded the maximum age of {MaxAge} specified by the 'max-age' cache directive. " +
+        "However, the 'max-stale' cache directive was specified without an assigned value and a stale response of any age is accepted.",
+        EventName = "ExpirationInfiniteMaxStaleSatisfied")]
+    internal static partial void ExpirationInfiniteMaxStaleSatisfied(this ILogger logger, TimeSpan age, TimeSpan maxAge);
+
+    [LoggerMessage(30, LogLevel.Debug, "The response time of the entry is {ResponseTime} and has exceeded its expiry date.",
+        EventName = "ExpirationExpiresExceeded")]
+    internal static partial void ExpirationExpiresExceeded(this ILogger logger, DateTimeOffset responseTime);
```

Comment:
This shares a name with another event (there is an overload that takes expiry). I know there is also the ID, but to avoid confusion I think it is best if logs have unique names.

Possibly combine the log methods? Or rename one of them.